__device-tests__/gutenberg-editor-sanity-test-4-visual.test.js

```diff
@@ -189,5 +231,169 @@ describe( 'Gutenberg Editor - Test Suite 4', () => {
 
 			await editorPage.removeBlock();
 		} );
+
+		it( 'Check if in DarkMode all components gets proper colors', async () => {
+			// Toggling dark mode
+			await toggleDarkMode( editorPage.driver, true );
+			// The Android editor requires a restart to apply dark mode
+			if ( isAndroid() ) {
+				await editorPage.driver.resetApp();
+				await isEditorVisible( editorPage.driver );
+			}
+
+			// Add Buttons block
+			await editorPage.addNewBlock( blockNames.buttons );
+			const buttonTextInput = await editorPage.getButtonBlockTextInputAtPosition();
+			await editorPage.typeTextToTextBlock(
+				buttonTextInput,
+				e2eTestData.shortButtonText
+			);
+
+			// Open and wait for block settings
+			await editorPage.openBlockSettings();
+			await editorPage.driver.sleep( 500 );
+
+			// Visual test check
+			const screenshot = await takeScreenshot();
+			expect( screenshot ).toMatchImageSnapshot();
+
+			// Clean up test
+			await editorPage.dismissBottomSheet();
+			await editorPage.moveBlockSelectionUp();
+			await editorPage.removeBlock();
+			await toggleDarkMode( editorPage.driver, false );
+			// The Android editor requires a restart to apply dark mode
+			if ( isAndroid() ) {
+				await editorPage.driver.resetApp();
+				await isEditorVisible( editorPage.driver );
+			}
+		} );
+	} );
+
+	describe( 'Group block', () => {
+		it( 'navigates nested structure', async () => {
+			// Add nested structure
+			await editorPage.setHtmlContent( e2eTestData.groupNestedStructure );
+
+			// Tap on the bottom-most block in hierarchy to check that navigation
+			// down works according to deepest-descendent-first approach.
+			const textXPath = isAndroid()
+				? `//android.widget.Button[@content-desc="${ blockNames.paragraph } Block. Row 1. Level ${ GROUP_NESTED_STRUCTURE_LEVELS }"]//android.widget.EditText`
+				: `(//*[@name="${ blockNames.paragraph } Block. Row 1. Level ${ GROUP_NESTED_STRUCTURE_LEVELS }"])[last()]`;
```

Comment:
I tried using the same approach as on Android querying a text element (i.e. `XCUIElementTypeTextView`). However, this doesn't work for text elements being nested beyond two levels, for some reason they are not rendered as `XCUIElementTypeTextView`.